### DISCUSSION TITLE
Type errors using realtime presence

### DISCUSSION THREAD

**Discussion by BarkhatovAndrew** [supabase/supabase#11333] [Questions]
I'm trying to implement online users checking using supabase realtime. I found this example in documentation https://supabase.com/docs/guides/realtime/quickstart
But I got some type errors in this code:

```
const channel = supabase.channel('online-users', {
    config: {
      presence: {
        key: data.user.username
      }
    }
  })

channel.on('presence', { event: 'sync' }, () => {
  players = Object.keys(channel.presenceState())
})

channel.subscribe(async (status) => {
  if (status === 'SUBSCRIBED') {
    await channel.track({ online_at: new Date().toISOString() })
  }
})
```

Error: Argument type "presence" is not assignable to parameter type `${REALTIME_LISTEN_TYPES.BROADCAST}` 

So how can I fix it? Broadcast not working in this case, and I need presence.
By the way I think strange that it require `${TYPE}` instead of just TYPE.

**Comment by w3b6x9** [comment]
@BarkhatovAndrew I'll take a look at this in `realtime-js`.
